src/transport_ofi.h

```diff
@@ -286,30 +288,30 @@ void shmem_transport_put_quiet(void)
     }
 
     /* wait for put counter to meet outstanding count value */
-#ifdef ENABLE_COMPLETION_POLLING
-    uint64_t success, fail;
-    do {
+    uint64_t success = 0, fail, poll_count = 0;
+    while (poll_count < shmem_transport_ofi_put_poll_limit && \
+           success < shmem_transport_ofi_pending_put_counter) {
         success = fi_cntr_read(shmem_transport_ofi_put_cntrfd);
-        fail = fi_cntr_readerr(shmem_transport_ofi_put_cntrfd);
+        fail    = fi_cntr_readerr(shmem_transport_ofi_put_cntrfd);
 
         if (success < shmem_transport_ofi_pending_put_counter && fail == 0) {
             SPINLOCK_BODY();
+        } else if (fail) {
+            struct fi_cq_err_entry e = {0};
+            fi_cq_readerr(shmem_transport_ofi_put_nb_cqfd, (void *)&e, 0);
+            OFI_CQ_ERROR(shmem_transport_ofi_put_nb_cqfd, &e);
         }
-        else if (fail) {
+        poll_count++;
+    }
+    if (success < shmem_transport_ofi_pending_put_counter) {
```

Comment:
That is, if a put is issued by another thread after the above polling loop and before this check, it will cause the thread that was trying to exit to instead call ```fi_cntr_wait```.  This will be an error if the counter has ```FI_WAIT_NONE``` set.